chore: remove Python 3.8 references

Faraz32123 opened the pull request:
Achieves part of https://github.com/overhangio/tutor/issues/1129.
- Update python 3.8 references to 3.9
- Use Github actions to use 3.9
- Ensure tests are working as expected on 3.9

DawoudSheraz commented on changelog.d/20241111_173552_faraz.maqsood_remove_py38_references.md:
```diff
@@ -0,0 +1 @@
+- 💥 [Deprecation] Drop support for python 3.8 and set Python 3.9 as the minimum supported python version. (by @Farz32123)
```
nit: incorrect GitHub username

Commit:
chore: remove Python 3.8 references
- drop support for python 3.8 and set Python 3.9 as the minimum supported python version.
